src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -196,6 +196,40 @@ final case class AgreementApiServiceImpl(
     }
   }
 
+  /**
+    * Code: 200, Message: Agreement updated., DataType: Agreement
+    * Code: 404, Message: Agreement not found, DataType: Problem
+    * Code: 400, Message: Invalid ID supplied, DataType: Problem
+    */
+  override def updateAgreementById(agreementId: String, agreementUpdatePayload: AgreementUpdatePayload)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerAgreement: ToEntityMarshaller[Agreement]
+  ): Route = {
+    val result = for {
+      agreementUUID <- agreementId.toFutureUUID
+      agreement     <- agreementManagementService.getAgreementById(agreementUUID)
+      seed = AgreementManagement.UpdateAgreementSeed(
```

Comment:
This update must be granted only to draft agreements, so I suggest you add a control here.